implementations/c/include/ockam/transport.h

```diff
@@ -0,0 +1,154 @@
+/**
+ ********************************************************************************************************
+ * @file        transport.h
+ * @brief       Public-facing API function prototypes for Ockam's transport library
+ ********************************************************************************************************
+ */
+#ifndef OCKAM_TRANSPORT_H
+#define OCKAM_TRANSPORT_H 1
+
+/*
+ ********************************************************************************************************
+ *                                             INCLUDE FILES                                            *
+ ********************************************************************************************************
+ */
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/tcp.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <errno.h>
+#include <time.h>
```

Comment:
These includes look like they tightly couple the transport interface to a platform. 
Which ones here do we really need in `transport.h`?

`stdlib.h` and `error.h` make sense, although we could debate if we need to decouple from `stdlib.h`.

Reply:
I moved all but stdlib.h, string.h, and stdint.h into the posix-specific include file in the transport source tree